### PR TITLE
Migrate delta streaming to node-fetch

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/nylas/nylas-nodejs/issues/174 https://github.com/nylas/nylas-nodejs/pull/214

Another migration example with node streams. Also added abort-controller polyfill and getRequest analog of requestOptions method in nylas connection

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by TrySound** [comment]
cc @sdwvit @Gerdie

**Comment by TrySound** [comment]
@sdwvit And this one?

File: __tests__/delta-spec.js [hunk 1]
```diff
@@ -16,18 +25,7 @@ describe('Delta', () => {
...
-      request.abort = jest.fn();
-      return request;
-    };
-
-    const createResponse = statusCode => {
-      const response = new PassThrough();
-      response.statusCode = statusCode;
-      return response;
-    };
+    config.setApiServer('http://nylas.com');
```
Comment (sdwvit) [inline __tests__/delta-spec.js:28]:
```suggestion
    config.setApiServer('https://nylas.com');
```
?

Comment (sdwvit) [reply __tests__/delta-spec.js:28]: or i guess it doesn't matter much here

**Review by sdwvit — APPROVED** [review]

**Pull request merged**